--- src/common/math.h.orig	2023-06-03 17:01:51.238602352 +0000
+++ src/common/math.h	2023-06-03 23:10:56.501521879 +0000
@@ -19,7 +19,15 @@
 #pragma once
 
 #include <stddef.h>
+#ifdef __cplusplus
+#include <cmath>
+using std::isnan;
+using std::isinf;
+using std::isfinite;
+using std::isnormal;
+#else
 #include <math.h>
+#endif
 #include <stdint.h>
 #include "common/sse.h"		// also includes darktable.h
 
